src/common/ina.rs

```diff
@@ -123,19 +118,15 @@ impl fmt::Display for Model {
     }
 }
 
-pub struct Ina<H> {
+struct Ina<H: I2CHandle> {
```

Comment:
I'm still a little puzzled here, and I think it is mostly my issue re rust comprehension,  but how can `H` ever end up being something other than `I2cHandleType` here? The only caller of `Ina<H>::new` is `from_config`, and it always calls it with the result of calling `Board::get_i2c_by_name`, which returns `pub type I2cHandleType = Arc<Mutex<dyn I2CHandle + Send>>;`, which happens to *also* be an `I2CHandle`. Why can't the `i2c_handle` field just be declared to be of type `I2cHandleType`?

Reply:
it can be, I wanted to be able to support any implementation of the `I2CHandle` trait, but now that nothing is public anymore you're correct.

Reply:
resolved offline